pkg/api/v1/pod/util.go

```diff
@@ -63,21 +63,30 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 
 // TODO: remove this function when init containers becomes a stable feature
 func SetInitContainersAndStatuses(pod *v1.Pod) error {
-	var initContainersAnnotation string
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersAnnotationKey]
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersBetaAnnotationKey]
-	if len(initContainersAnnotation) > 0 {
+	// If there is a beta annotation, copy to alpha key.
+	// See commit log for PR #31026 for why we do this.
+	if valueBeta, okBeta := pod.Annotations[v1.PodInitContainersBetaAnnotationKey]; okBeta {
+		pod.Annotations[v1.PodInitContainersAnnotationKey] = valueBeta
```

Comment:
I haven't looked at this PR yet.

The problem we discussed was that old node will ignore beta annotation and send it back.
 New apiserver will use the empty beta annotation when it presents.